src/hash.cpp

```diff
@@ -98,10 +116,14 @@ std::string md5::hex_digest() const
 
 sha1::sha1(const std::string& str)
 {
+#if !defined(__IPHONEOS__)
```

Comment:
Would perfer if you kept the formatting consistent and used `#ifndef __IPHONEOS__` here like you do in the next file.